spec/models/internal/event_spec.rb

```diff
@@ -0,0 +1,192 @@
+require "rails_helper"
+
+describe Internal::Event do
+  describe "attributes" do
+    it { is_expected.to respond_to :id }
+    it { is_expected.to respond_to :name }
+    it { is_expected.to respond_to :summary }
+    it { is_expected.to respond_to :description }
+    it { is_expected.to respond_to :building }
+    it { is_expected.to respond_to :is_online }
+    it { is_expected.to respond_to :start_at }
+    it { is_expected.to respond_to :end_at }
+    it { is_expected.to respond_to :provider_contact_email }
+    it { is_expected.to respond_to :provider_organiser }
+    it { is_expected.to respond_to :provider_target_audience }
+    it { is_expected.to respond_to :provider_website_url }
+  end
+
+  describe "validations" do
+    let(:now) { Time.zone.now }
+
+    describe "#name" do
+      it { is_expected.to allow_value("test").for :name }
+      it { is_expected.to_not allow_values("", nil).for :name }
+    end
+
+    describe "#summary" do
+      it { is_expected.to allow_value("test").for :summary }
+      it { is_expected.to_not allow_values("", nil).for :summary }
+    end
+
+    describe "#description" do
+      it { is_expected.to allow_value("test").for :description }
+      it { is_expected.to_not allow_values("", nil).for :description }
+    end
+
+    describe "#is_online" do
+      it { is_expected.to allow_values(true, false).for :is_online }
+      it { is_expected.to_not allow_value(nil).for :is_online }
+    end
+
+    describe "#start_at" do
+      it { is_expected.to_not allow_values(nil, now - 1.minute, now).for :start_at }
+      it { is_expected.to allow_value(Time.zone.now + 1.minute).for :start_at }
+    end
+
+    describe "#end_at" do
+      it { is_expected.to_not allow_values(nil, now - 1.minute, now).for :end_at }
+      it { is_expected.to allow_value(now + 1.minute).for :end_at }
+
+      it "is expected not to allow end at to be the same as start at" do
+        subject.start_at = now + 1.minute
+        is_expected.to_not allow_value(now + 1.minute).for :end_at
```

Comment:
`is_expected.to_not allow_value(subject.start_at).for :end_at`, take it or leave it